experanto/data.py

```diff
@@ -30,3 +32,49 @@ def __getitem__(self, idx):
         times = self._sample_times[s : s + self.chunk_size]
         data, _ = self._experiment.interpolate(times)
         return self.DataPoint(*list(data.values()))
+
+
+class Mouse2pStaticImageDataset(Dataset):
+    def __init__(self, root_folder: str, tier: str, offset=100) -> None:
+        self.root_folder = Path(root_folder)
+        self.tier = tier
+        self.offset = offset
+        self._experiment = Experiment(root_folder)
+        self.device_names = self._experiment.device_names
+        self.DataPoint = namedtuple("DataPoint", self.device_names)
+        self._read_trials()
+    
+    def _read_trials(self):
+        screen = self._experiment.devices["screen"]
+        self._trials = [
+            t for t in screen.trials
+            if isinstance(t, ImageTrial) and t.get_meta("tier") == self.tier
+        ]
+        s_idx = np.array([t.first_frame_idx for t in self._trials])
+        if len(s_idx):
+            self._start_times = screen.timestamps[s_idx]
+            end_times = screen.timestamps[s_idx + 1]
+            self._trial_duration = np.min(end_times - self._start_times)
+            assert np.all(
+                end_times - self._start_times - self._trial_duration < 0.1), \
+                "Trial durations vary by more than 100 ms"
```

Comment:
we should make this assumption a bit more explicit somewhere. I think there might be some experiments where the stimulus duration was more unequal.

Now that I think about it - maybe this has to be a user specified argument. Offset + duration of time window could be user specified. If we add an assert as explicit data sanity check, then we need too many Dataset classes for all kinds of experiments.